public/editor/scripts/project-list.js

```diff
@@ -30,16 +37,57 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
     return "{{ momentJSLastEdited | safe }}".replace("<% timeElapsed %>", timeElapsed);
   }
 
+  function storageAvailable(type) {
+    try {
+      var storage = window[type],
+      x = '__storage_test__';
+      storage.setItem(x, x);
+      storage.removeItem(x);
+      return true;
+    }
+    catch(e) {
+      return false;
+    }
+  }
+ 
+  var favoriteArray = [];
+
   Array.prototype.forEach.call(projects, function(project) {
     var projectSelector = "#" + project.getAttribute("id");
     var lastEdited = project.getAttribute("data-project-date_updated");
 
+    if(storageAvailable('localStorage')) {
+      var projectId = project.getAttribute("data-project-id");
+      if(favorites.indexOf(projectId) != -1){
+        favoriteArray.push(project);
+        $(projectSelector + " .project-favorite-button").toggleClass("project-unfavorite-button");
+      }
+      $(projectSelector + " .project-favorite").on("click", function() {
+        favorites = JSON.parse(localStorage.getItem('project-favorites'));
+        if(favorites === null){
+          favorites = [];
+        }
+        if(favorites.indexOf(projectId) == -1) {
+          favorites.push(projectId);
+          localStorage.setItem('project-favorites', JSON.stringify(favorites));
+          $(projectSelector + " .project-favorite-button").toggleClass("project-unfavorite-button");
+        }
+        else {
+          favorites.splice(favorites.indexOf(project.getAttribute("data-project-id")), 1);
```

Comment:
In fact, let's store the value of `favorites.indexOf(projectId)` in a variable above the `if...else` block so that we can simply do:
```js
var indexOfProjectInFavorites = favorites.indexOf(projectId);

if(indexOfProjectInFavorites === -1) {
   ...
} else {
   favorites.splice(indexOfProjectInFaavorites, 1);
   ...
}